x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlSpMetadataBuilder.java

```diff
@@ -225,7 +225,7 @@ public EntityDescriptor build() throws Exception {
             spRoleDescriptor.getNameIDFormats().add(buildNameIdFormat());
         }
         spRoleDescriptor.getAssertionConsumerServices().add(buildAssertionConsumerService());
-        if (attributeNames.size() > 0) {
+        if (attributeNames != null && attributeNames.size() > 0) {
```

Comment:
I don't think we need to change this any more since we initialize `attributesNames` properly in the single ctor of this class.